### PR TITLE
Fix top-parent menu not getting closed by action-item click

### LINKED ISSUE #2965: Clicks in nested context menus don't close the menu correctly
When clicking an item in a sub-menu, the whole menu tree should close.

https://streamable.com/3lv90

### PR DESCRIPTION
**Pull request by iiSaLMaN**
- [x] Depends on #2987 

Closes #2965

### CODE REVIEW

File: osu.Framework/Graphics/UserInterface/Menu.cs [hunk 1]
```diff
@@ -523,7 +523,7 @@ private void closeFromChild(MenuItem source)
         {
             if (IsHovered || (parentMenu?.IsHovered ?? false)) return;
 
-            if (triggeringItem?.Items?.Contains(source) ?? false)
+            if (triggeringItem?.Items?.Contains(source) ?? triggeringItem == null)
```
Comment (bdach) [inline osu.Framework/Graphics/UserInterface/Menu.cs]: Why is this not `?? true` instead at this point?

Comment (iiSaLMaN) [reply osu.Framework/Graphics/UserInterface/Menu.cs]: we might have a triggering item that has no own items, making `source` useless i suppose.

**Comment by iiSaLMaN** [comment]
> I don't quite understand why `TestSceneNestedMenus` closes the topmost menu, and `TestSceneClosableMenu` doesn't.

wait.., isn't it supposed to be the other way around?

**Comment by bdach** [comment]
Yeah, I mixed them up, my bad.

**Comment by iiSaLMaN** [comment]
`TestSceneNestedMenus` has a top-level menu which should never be closed (like horizontal menu bar on most of the programs), while `TestSceneClosableMenu` doesn't. (like context menus)

**Pull request merged**